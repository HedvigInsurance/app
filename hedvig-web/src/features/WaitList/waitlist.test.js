import React from "react"
import { mount } from "enzyme"
import ReactRouterEnzymeContext from "react-router-enzyme-context"

import { WaitListComponent as WaitList } from "./index"

describe('<WaitList />', () => {
  it('Should fetch waitlist position when mounting', () => {
    const options = new ReactRouterEnzymeContext()
    const spy = jest.fn()
    mount(
      <WaitList
        fetchWaitlistPosition={spy}
        match={{params: {id: "some-uuid"}}}
      />,
      options.get()
    )

    expect(spy).toHaveBeenCalledTimes(1)
  })
})
