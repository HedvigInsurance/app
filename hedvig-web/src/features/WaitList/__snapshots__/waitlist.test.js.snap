// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WaitList /> Should render correctly in status ACCESS 1`] = `
<main
  className="WaitList"
>
  <Header
    headerRight={<UNDEFINED />}
    static={true}
  />
  <article
    className="pure-g pure-centered"
  >
    <div
      className="pure-u-1-1 pure-u-lg-4-5 WaitList__content"
    >
      <div>
        <p
          className="WaitList__paragraph"
        >
          Väntan är över
        </p>
        <h1
          className="WaitList__header"
        >
          Välkommen till Hedvig
        </h1>
        <p
          className="WaitList__paragraph"
        >
          Din aktiveringskod är
        </p>
        <p
          className="WaitList__code"
        >
          APPLE123
        </p>
        <CopyToClipboard
          text="APPLE123"
        >
          <WhiteRoundedButton
            style={
              Object {
                "width": "200px",
              }
            }
          >
            Kopiera koden
          </WhiteRoundedButton>
        </CopyToClipboard>
        <p
          className="WaitList__paragraph"
        >
          och gå sedan
        </p>
        <a
          href="hedvig://"
          rel="noopener noreferrer"
          target="_blank"
        >
          <TurquoiseRoundedButton
            style={
              Object {
                "fontSize": "18px",
                "marginBottom": "4em",
                "width": "200px",
              }
            }
          >
            Till appen
          </TurquoiseRoundedButton>
        </a>
      </div>
    </div>
  </article>
</main>
`;

exports[`<WaitList /> Should render correctly in status NOT_FOUND 1`] = `
<main
  className="WaitList"
>
  <Header
    headerRight={<UNDEFINED />}
    static={true}
  />
  <article
    className="pure-g pure-centered"
  >
    <div
      className="pure-u-1-1 pure-u-lg-4-5 WaitList__content"
    >
      <WaitListError />
    </div>
  </article>
</main>
`;

exports[`<WaitList /> Should render correctly in status WAITLIST 1`] = `
<main
  className="WaitList"
>
  <Header
    headerRight={<UNDEFINED />}
    static={true}
  />
  <article
    className="pure-g pure-centered"
  >
    <div
      className="pure-u-1-1 pure-u-lg-4-5 WaitList__content"
    >
      <div>
        <h1
          className="WaitList__header"
        >
          Snart är det dags!
        </h1>
        <div
          className="pure-g pure-centered"
        >
          <div
            className="pure-u-1-1 pure-u-md-3-5 pure-u-lg-5-8 pure-u-xl-1-2"
          >
            <div
              className="WaitList__position-box"
            >
              <p
                className="WaitList__paragraph WaitList__paragraph--marginless"
              >
                Före dig på väntelistan står
              </p>
              <h3
                className="WaitList__header WaitList__heading--position"
              >
                1
                 
                <span
                  className="WaitList__person-label"
                >
                  personer
                </span>
              </h3>
            </div>
          </div>
        </div>
        <p
          className="WaitList__paragraph"
        >
          Du får en aktiveringskod på mailen så fort det är din tur!
        </p>
        <p
          className="WaitList__paragraph"
        >
          Vi ses snart!
        </p>
      </div>
    </div>
  </article>
</main>
`;
